### ISSUE TITLE
Authorization not correctly working for stitched, local schemas

### ISSUE THREAD

**Issue by barticus** [ChilliCream/hotchocolate#1273]
**Describe the bug**
When using schema stitching, and adding built schemas (i.e. not over HTTP) to the schema stitching builder, the authorization middleware/directive that is registered with the sub-schemas is not applied correctly, and causes the AUTH_NOT_AUTHENTICATED error to be thrown in all cases (with auth, without auth, with correct role/policy, without correct role/policy).

**To Reproduce**
1. Build 2 distinct schemas similar to the following:
```
var schema1 = HotChocolate.SchemaBuilder.New()
    .AddDocumentFromString(schema1Content)
    .AddAuthorizeDirectiveType()
    .BindComplexType<GraphQL.Schema1.Query>(t => t.To("Query"))
    .BindComplexType<GraphQL.Schema1.Mutation>(t => t.To("Mutation"))
    .Create();
var schema2 = HotChocolate.SchemaBuilder.New()
    .AddDocumentFromString(schema2Content)
    .AddAuthorizeDirectiveType()
    .BindComplexType<GraphQL.Schema2.Query>(t => t.To("Query"))
    .BindComplexType<GraphQL.Schema2.Mutation>(t => t.To("Mutation"))
    .Create();
```
2. Stitch the schemas together:
```
services.AddStitchedSchema(builder => builder
    .AddSchema("schema1", schema1)
    .AddSchema("schema2", schema2);
```
3. Make sure you are using the authorize directive in one of the schemas, e.g.
```
type Profile {
  email: String!
  firstName: String!
  id: Int!
  lastName: String!
}
type Query {
  profile: Profile @authorize
}
```
4. Make sure you are using `services.AddAuthorization(...)` and `services.AddAuthentication(...)` in Startup's `ConfigureServices`, and `app.UseAuthentication();` in `Configure` (before the call to `app.UseGraphQLHttpPost(...)`)

**Expected behavior**
Authorization to be correctly applied.

**Desktop (please complete the following information):**
 - Framework: ASP.NET Core 2.2 (netcoreapp2.2)
 - HotChocolate Packages, all version 10.2.0:
"HotChocolate"
"HotChocolate.Stitching"
"HotChocolate.AspNetCore"
"HotChocolate.AspNetCore.Playground"
"HotChocolate.AspNetCore.Authorization"

**Additional context**

1. Changing `services.AddStitchedSchema(...)` line above to `services.AddGraphQL(schema1)` DOES apply the authorization flow as expected, it seems to be only when combined with stitching that this issue presents itself
2. I've added a `QueryRequestInterceptor` and logged out the context.User which is correctly set from the asp.net authorization flow
3. I've added middleware in `AddSchemaConfiguration` to explicitly set `context.ContextData[nameof(ClaimsPrincipal)]` which appears to be the field the directive middleware uses, but it still throws the same error.

I'm really liking HotChocolate so far, however I'm struggling to get this piece in particular working. I might be "using it wrong" so please feel free to correct me. I could create a repro of this in a small repo, please let me know if this isn't clear (or clearly the wrong approach) and i can do this.
I've tried to find an existing issue around this and must say, your team has been great with getting back to everyone and supporting the community. 
This partly relates to tickets like #1231 . I also saw mentioned in #1228 that theres an "interceptor" i might need to have configured, but couldn't find further information on that.

**Comment by michaelstaib** [comment]
Hi @barticus, 

thanks for this detailed issue. The cause for this is that the authorise directive need the claims principal in the execution context.

When stitching over HTTP you would basically pass on an auth token or create a new auth token to the remote schema in and then the claims principal would be associated with the execution context.

In the case when we use in-memory schemas this does not happen. 

I will have to check how we can enable this, I honestly did not think about this scenario so far :)

**Comment by barticus** [comment]
Thanks @michaelstaib for the prompt response.
I thought it might be something along those lines. I could see examples for passing along the headers for HTTP stitching, but couldn't see any notes on whether there was extra work for local schemas.
As mentioned, I had a go at writing some custom middleware/request filler to pass the user along, but couldn't get it working. I could write my own authorizer directive that loads the user from the http context accessor, but just wasn't sure if I was just heading down the wrong path.
Cheers.

**Comment by thorgeir** [comment]
Hi, I'm also running into this problem. Is there a workaround or a plan to fix the issue?

**Comment by michaelstaib** [comment]
Not yet. But a fix is planed ... I am marking this as a bug since we already know what it is.

**Comment by barticus** [comment]
As a work around, I copied the code for the authorize directive and changed it so it was using HttpContextAccessor rather than trying to get the principal from the context.
I would post that here but I think in the end I changed tactics and decided not to use stitching and just use multiple calls to "AddDocumentFromFile" as I wasn't trying to do anything complicated that needed stitching (and it is more for schemas over http, not local)

**Comment by benmccallum** [comment]
Following as I have the same issue.

**Comment by benmccallum** [comment]
@michaelstaib, believe this has been fixed on 10.5.0-preview.9 if you want to mark it as such.

**Comment by michaelstaib** [comment]
This is fixed in preview 9... lets close the issue :)
